src/librustc/middle/lang_items.rs

```diff
@@ -327,6 +327,8 @@ language_item_table! {
     PanicBoundsCheckFnLangItem,      "panic_bounds_check",      panic_bounds_check_fn;
     PanicFmtLangItem,                "panic_fmt",               panic_fmt;
 
+    // ExchangeMallocFnLangItem cannot unwind, or MIR trans will break. See note
+    // on `malloc_raw_dyn` in librustc_trans/base.rs.
```

Comment:
You can simply change how MIR for allocation is generated. No need to fiddle in trans, unless there’s some old cruft in old trans that still generates calls to this itself… in which case you’ll have a hard and bad time fixing this :)